src/main/java/com/couchbase/lite/storage/SQLiteStorageEngineBase.java

```diff
@@ -0,0 +1,303 @@
+        try {
+            database.execSQL(sql);
+        } catch (com.couchbase.lite.database.SQLException e) {
+            throw new SQLException(e);
+        }
+    }
+
+    @Override
+    public void execSQL(String sql, Object[] bindArgs) throws SQLException {
+        try {
+            database.execSQL(sql, bindArgs);
+        } catch (com.couchbase.lite.database.SQLException e) {
+            throw new SQLException(e);
+        }
+    }
+
+    @Override
+    public Cursor rawQuery(String sql, String[] selectionArgs) {
+        return new SQLiteCursor(database.rawQuery(sql, selectionArgs));
+    }
+
+    @Override
+    public long insert(String table, String nullColumnHack, ContentValues values) {
+        return database.insert(table, nullColumnHack, toContentValues(values));
+    }
+
+    @Override
+    public long insertOrThrow(String table, String nullColumnHack, ContentValues values) throws SQLException {
+        try {
+            return database.insertOrThrow(table, nullColumnHack, toContentValues(values));
+        } catch (com.couchbase.lite.database.SQLException e) {
+            if(e instanceof com.couchbase.lite.database.sqlite.exception.SQLiteConstraintException)
+                throw new SQLException(SQLException.SQLITE_CONSTRAINT, e);
+            else
+                throw new SQLException(e);
+        }
+    }
+
+    @Override
+    public long insertWithOnConflict(String table, String nullColumnHack,
+                                     ContentValues initialValues, int conflictAlgorithm) {
+        return database.insertWithOnConflict(table, nullColumnHack,
+                toContentValues(initialValues), conflictAlgorithm);
+    }
+
+    @Override
+    public int update(String table, ContentValues values, String whereClause, String[] whereArgs) {
+        return database.update(table, toContentValues(values), whereClause, whereArgs);
+    }
+
+    @Override
+    public int delete(String table, String whereClause, String[] whereArgs) {
+        return database.delete(table, whereClause, whereArgs);
+    }
+
+    @Override
+    public void close() {
+        database.close();
+    }
+
+    @Override
+    public boolean supportEncryption() {
+        if (isSupportEncryption == null) {
+            boolean support = SQLiteDatabase.isSupportEncryption();
+            isSupportEncryption = new AtomicBoolean(support);
+        }
+        return isSupportEncryption.get();
+    }
+
+    @Override
+    public byte[] derivePBKDF2SHA256Key(String password, byte[] salt, int rounds) {
+        if (!supportEncryption())
+            return null;
+        return Key.derivePBKDF2SHA256Key(password, salt, rounds);
+    }
+
+    @Override
+    public String toString() {
+        return "SQLiteStorageEngine {" +
+                "database=" + Integer.toHexString(System.identityHashCode(database)) +
+                "}";
+    }
+
+    private com.couchbase.lite.database.ContentValues toContentValues(ContentValues values) {
```

Comment:
I copied from the old code and I didn't change it. Agree that this could be eliminated now. I will make changes.